src/Utilities/ArrayIntersector.php

```diff
@@ -22,14 +22,14 @@ class ArrayIntersector
     /**
      * the array with which the object was constructed, with all its keys exchanged with their associated values
      *
-     * @var (int|string)[]
+     * @var array<int|string, int|string>
```

Comment:
`array-key` rather than `int|string` could be used for these, though I don't know if one or the other is any clearer.

Reply:
Like this?
`array<array-key, array-key>`

Reply:
I was thinking `array<array-key, array-key>` - it's mentioned in the DocBlock for the class that the values have to be integers or strings becuase `array_flip` is used - though put another way the values have to be allowed as array keys for this reason, which in turn means they have to be integers or strings because those are the only types allowed for array keys.

But like I said, I don't know whether one or other is any clearer.

Reply:
Then let's settle for `<array-key, array-key>` for the time being. We can change it later if anything else seems to make more sense.